extension/js/common/browser/browser.ts

```diff
@@ -23,33 +22,21 @@ export class Browser {
     return buf;
   }
 
+  // Borrowed from https://dev.to/nombrekeff/download-file-from-blob-21ho
   public static saveToDownloads = (attachment: Attachment) => {
     const blob = new Blob([attachment.getData()], { type: attachment.type });
-    if (window.navigator && window.navigator.msSaveOrOpenBlob) {
-      window.navigator.msSaveBlob(blob, attachment.name);
-    } else {
-      const a = window.document.createElement('a');
-      a.href = window.URL.createObjectURL(blob);
-      a.download = Xss.escape(attachment.name);
-      if (typeof a.click === 'function') { // tslint:disable-line:no-unbound-method - only testing if exists
-        a.click();
-      } else { // safari
-        const ev = document.createEvent('MouseEvents');
-        // @ts-ignore - safari only. expected 15 arguments, but works well with 4
-        ev.initMouseEvent('click', true, true, window);
-        a.dispatchEvent(ev);
-      }
-      if (Catch.browser().name === 'firefox') {
-        try {
-          a.remove();
-        } catch (err) {
-          if (!(err instanceof Error && err.message === 'Node was not found')) {
-            throw err;
-          }
-        }
-      }
-      Catch.setHandledTimeout(() => window.URL.revokeObjectURL(a.href), 0);
-    }
+    // Create a link element
+    const a = window.document.createElement('a');
+    // Set link's href to point to the Blob URL
+    a.href = window.URL.createObjectURL(blob);
+    a.download = Xss.escape(attachment.name);
+    // Append link to the body
+    document.body.appendChild(a);
+    // Dispatch click event on the link
+    // This is necessary as link.click() does not work on the latest Firefox
+    a.dispatchEvent(new MouseEvent('click', { bubbles: true, cancelable: true, view: window }));
+    // Remove link from body
+    document.body.removeChild(a);
```

Comment:
Just double checking - does this also work on all of recent Chrome, Safari, Firefox and Edge?

Reply:
Because we reuse this code on the website too when sometimes other browsers access it. Sometimes the browsers are older.

Reply:
There actually was a small issue with the new approach, thanks for asking @tomholub 

I decided to revert the change and only remove the IE-related code that caused TS to complain. 
If it ain't broke, don't fix it (c)

Just in case tested manually in Chrome, Firefox and Edge, all is well.

Please review without whitespaces: https://github.com/FlowCrypt/flowcrypt-browser/pull/3458/files?w=1